### PR TITLE
Lcs in c

### LINKED ISSUE #1649: Longest Common Subsequence in C
As LCS is not available in C, I would like to work on it as part of GsSoc 2020. @singh-shreya6  @asha15 @raghav-dalmia kindly assign this to me

### PR DESCRIPTION
**Pull request by NVS16**
Fixes #1649 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- LCS for C language added

#### Languages Used:
- C

#### Files Added:
- Longest_Common_Subsequence.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/*
+
+Sample Input: 
+7 8
+10 15 20 25 30 35 40
+10 12 23 25 28 30 32 40
+
+Sample Output: 
+4
+*/
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Add a new line at end of file

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Please add problem description at the start of code as multi line comment.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+// Function that computes the longest commmon subsequence between two arrays of numbers
+// Input: First Array, Second Array, Size of First Array, Size of Second Array
+// Output: Longest Common Subsequnce
+int LongestCommonSubsequence(int arr1[], int arr2[], int size1, int size2) {
+    // Declaring a 2-D array LCS where LCS[i][j] denotes the longest common subsequence of
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Give a blank line here for greater readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+// Function that computes the longest commmon subsequence between two arrays of numbers
+// Input: First Array, Second Array, Size of First Array, Size of Second Array
+// Output: Longest Common Subsequnce
+int LongestCommonSubsequence(int arr1[], int arr2[], int size1, int size2) {
+    // Declaring a 2-D array LCS where LCS[i][j] denotes the longest common subsequence of 
+    // the two arrays until the ith element of the first array and the jth element of the second array
+    int LCS[size1 + 1][size2 + 1];
+    for (int i = 0; i <= size1; i++) {
+        for (int j= 0; j <= size2; j++) {
+            // Scenario 1: putting a 0 if either of the index is 0
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Give a blank line before this comment for greater readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            if (i == 0 || j == 0) {
+                LCS[i][j] = 0;
+            }             
+            // Scenario 2: ith element of the first array is equal to jth element of array of second array
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Give a blank line before this comment for greater readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            else if (arr1[i - 1] == arr2[j - 1]) {
+                LCS[i][j] = LCS[i - 1][j - 1] + 1;
+            }             
+            // Scenario 3: ith element of the first array is not equal to jth element of array of second array.
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Give a blank line before this comment for greater readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.c [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            // In this case we chose the max between LCS[i - 1][j] and LCS[i][j - 1]
+            else {
+                LCS[i][j] = LCS[i - 1][j] > LCS[i][j - 1] ? LCS[i - 1][j] : LCS[i][j - 1];
```
Comment (SakshiShreya) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.c]: Break the line. You should not have more than 80 characters in one line.

**Comment by NVS16** [comment]
Hey, @singh-shreya6  I have added the requested changes. You can review them.

**Comment by NVS16** [comment]
Have removed the a.out file.

**Pull request merged**